backend/Origam.DA.Service/ReferenceIndexManager.cs

```diff
@@ -132,7 +150,14 @@ private static void GetReferencesFromText(AbstractSchemaItem item)
             }
         }
     }
-
+    private static void Addreference(AbstractSchemaItem retrievedObj)
+    {
+        GetReferencesFromDependencies(retrievedObj);
+        foreach (AbstractSchemaItem item in retrievedObj.ChildItems)
```

Comment:
Is it necessary to loop through all the child items? I see this method is only called when the InstancePersisted event is raised in FilePersistenceProvider. 
https://github.com/origam/origam/blob/c895d6e9327e9853adbe0acc78e6ffe16eb5c16f/backend/Origam.DA.Service/FilePersistenceProvider.cs#L87

And that is raised on all children when any AbstractItem is peristed.

https://github.com/origam/origam/blob/c895d6e9327e9853adbe0acc78e6ffe16eb5c16f/backend/Origam.Schema/AbstractSchemaItem.cs#L353

I'm just curious since this is the reason why I removed it...

Reply:
Action Wizards in Architect create a new object with child items in one time and save it. FilePersistenceProvider  adds into ReferenceIndex only parent not childs. Childs are too important.

Reply:
ok